internal/handlers/pg/pgdb/indexes_test.go

```diff
@@ -39,32 +39,69 @@ func TestCreateIndexIfNotExists(t *testing.T) {
 	collectionName := testutil.CollectionName(t)
 	setupDatabase(ctx, t, pool, databaseName)
 
-	indexName := "test"
-	err := pool.InTransaction(ctx, func(tx pgx.Tx) error {
-		idx := Index{
-			Name: indexName,
-			Key:  []IndexKeyPair{{Field: "foo", Order: types.Ascending}, {Field: "bar", Order: types.Descending}},
-		}
-		return CreateIndexIfNotExists(ctx, tx, databaseName, collectionName, &idx)
-	})
-	require.NoError(t, err)
-
-	tableName := collectionNameToTableName(collectionName)
-	pgIndexName := indexNameToPgIndexName(collectionName, indexName)
+	for name, tc := range map[string]struct {
+		expecteDefinition string // the expected index definition in postgresql
+		index             Index  // the index to create
+	}{
+		"keyWithoutNestedField": {
+			index: Index{
+				Name: "foo_and_bar",
+				Key: []IndexKeyPair{
+					{Field: "foo", Order: types.Ascending},
+					{Field: "bar", Order: types.Descending},
+				},
+			},
+			expecteDefinition: "((_jsonb -> 'foo'::text)), ((_jsonb -> 'bar'::text)) DESC",
+		},
+		"keyWithNestedField_level1": {
+			index: Index{
+				Name: "foo_dot_bar",
+				Key: []IndexKeyPair{
+					{Field: "foo.bar", Order: types.Ascending},
+				},
+			},
+			expecteDefinition: "(((_jsonb -> 'foo'::text) -> 'bar'::text))",
+		},
+		"keyWithNestedField_level2": {
+			index: Index{
+				Name: "foo_dot_bar_dot_c",
+				Key: []IndexKeyPair{
+					{Field: "foo.bar.c", Order: types.Ascending},
+				},
+			},
+			expecteDefinition: "((((_jsonb -> 'foo'::text) -> 'bar'::text) -> 'c'::text))",
+		},
+	} {
+		// https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721#how-to-solve-this
```

Comment:
```suggestion
```

I'm not sure we should mention this example here, as it's quite common for our codebase to have parallel subtests.